examples/3dtiles.html

```diff
@@ -134,58 +126,36 @@
             $3dTilesLayerRequestVolume.type = 'geometry';
             $3dTilesLayerRequestVolume.visible = true;
             $3dTilesLayerRequestVolume.sseThreshold = 1;
-            // add an event for have information when you move your mouse on a building
-            itowns.View.prototype.addLayer.call(view, $3dTilesLayerRequestVolume).then(function _() { window.addEventListener('mousemove', picking, false); })
-
-            // Add the UI Debug
-            var d = new debug.Debug(view, menuGlobe.gui);
-            debug.createTileDebugUI(menuGlobe.gui, view, view.wgs84TileLayer, d);
-            debug.create3dTilesDebugUI(menuGlobe.gui, view, $3dTilesLayerDiscreteLOD, d);
-            debug.create3dTilesDebugUI(menuGlobe.gui, view, $3dTilesLayerRequestVolume, d);
+            // Picking example - - - - - - - - - - -- - - - -- - - - -- - - - -- - - - -- - - - -- - - - -
 
 
-// Picking example - - - - - - - - - - -- - - - -- - - - -- - - - -- - - - -- - - - -- - - - -
-            function findBatchTableParent(object) {
-                if (object.batchTable) {
-                    return object.batchTable;
-                }
-                else if (object.parent) {
-                    return findBatchTableParent(object.parent);
-                } else {
-                    return undefined;
-                }
-            }
             function picking(event) {
                 var htmlInfo = document.getElementById('info');
+                var pickeds = event.pickeds;
                 htmlInfo.innerHTML = ' ';
-
-                const intersects = view.pickObjectsAt(
-                    event,
-                    $3dTilesLayerRequestVolume);
-
-                for (var i = 0; i < intersects.length; i++) {
-                    var interAttributes = intersects[i].object.geometry.attributes;
-                    if (interAttributes) {
-                        if (interAttributes._BATCHID) {
-                            var face = intersects[i].face.a;
-                            var batchID = interAttributes._BATCHID.array[face];
-                            var batchTable = findBatchTableParent(intersects[i].object);
-
-                            htmlInfo.innerHTML +='<li><b> Batch id: </b>'+ batchID +'</li>';
-                            Object.keys(batchTable).map(function(objectKey) {
-                                var value = batchTable[objectKey][batchID];
-                                // if the value is a integer or not
-                                var info = Number.isInteger(value) ? value.toString() : value.toFixed(3).toString();
-                                htmlInfo.innerHTML +='<li><b>' + objectKey.toString() + ': </b>'+ info +'</li>';
-                                return true;
-                            });
-
-                            return;
-                        }
-                    }
+                if (pickeds[0]) {
```

Comment:
So all this code just to remove the `view.pickObjectsAt`call, making things more opaque and slower in the process? (`defineProperty` makes things very slow)

Reply:
> So all this code just to remove the view.pickObjectsAtcall

Thank you for the encouragements :(
But if you didn't understand the interest of the PR, you should not answer like that

> more opaque 

As I said I have not done all the examples yet.

`eventView` unifies the compute of `pickObjectsAt`, but also `coordinate` and `viewCoords`
we can also add `viewEvent.features`

`eventView` also allows you to calculate once all the properties and especially not to recalculate them when there are several event calls. 

For example if the controls calculates a geographic coordinate under the mouse click, this calculation will be shared to the user in his event callback

Yes actually itowns make opaque a lot of code, it's a question of choice. And it's not your only opinion that decides what should be in or out of itowns.

> slower in the process

The 1 first `defineProperty` can be replace by `class` `eventViewer`

Reply:
Okay let's take it point by points, meta first:

> Yes actually itowns make opaque a lot of code, it's a question of choice. 

I'm not sure we use the word "opaque" in the same sense then. Opaque is never good for code, period. And it is even more the case when one have the ambition to develop a **framework**.

If it's a question of choice, then mine is all made and I'm not gonna r+ PRs that decrease code clarity if there is not a clear interest in doing so. I don't think it's the case here.

> And it's not your only opinion that decides what should be in or out of itowns.

Why the aggressivity? I have never pretended so, but as a reviewer, I do reviews. Something against that?

Now onto this PR:

>  But if you didn't understand the interest of the PR, you should not answer like that

I do understand what you want to do. I disagree with the current implementation of `view.on`. I disagree on creating a custom event class and adding an object creation in an event handler (which is perf problem n°2 btw) just to replace 2 lines of code (coords calculation + picking function calculation calls).

Is it that complicated to do:

```js
domElement.addEventListener('mousemove', () => {
    const vector = view.getPickingPositionFromDepth(eventToViewCoords(mouseEvent));
    // user then can create (or not!) coordinates from this Vector3. 
    // Or use it directly if they want to display an object for instance
});
```
? 
It's simpler, more flexible. Fits more with the fact we don't want to hide THREE.js logic, which is the differentiating factor of iTowns (or should be at least).

> eventView also allows you to calculate once all the properties and especially not to recalculate them when there are several event calls.

You do realize that your code is not doing that, right? Just to be sure :-)

> For example if the controls calculates a geographic coordinate under the mouse click, this calculation will be shared to the user in his event callback

In this case, this calculated value is specific to a control type, so belongs to the control API interface (and should be documented). The user can then register a framerequester to query this cached value at each frame. Your current code only works for the globe, and more precisely, for the globe with GlobeControls.

I do agree that some general helpers can be added (and `eventToViewCoords` was one of them). I do agree that we either need a shortcut to renderer.domElement, or a `view.on` shortcut. But in this case, view.on should just be:

```js
View.prototype.on = function on(event, cb) {
    view.mainLoop.gfxEngine.renderer.domElement.addEventListener(event, cb);
}
```
But I'd prefer a shortcut to the domElement (`view.domElement`? `view.canvas`?), because it's more "webby": people will understand they can use all the standard web stuff on this element and they won't have to rediscover a event api  custom to itowns.

I'll explain my idea about framerequester on the other comment.